test/distributed/org/apache/cassandra/distributed/test/JMXSslConfigDistributedTest.java

```diff
@@ -0,0 +1,205 @@
+            ImmutableMap<String, Object> encryptionOptionsMap = ImmutableMap.<String, Object>builder().putAll(validKeystore)
+                                                                            .put("enabled", true)
+                                                                            .put("accepted_protocols", Arrays.asList("TLSv1.2", "TLSv1.3", "TLSv1.1"))
+                                                                            .build();
+
+            try (Cluster cluster = builder().withNodes(1).withConfig(c -> {
+                c.with(Feature.JMX);
+                c.set("jmx_encryption_options", encryptionOptionsMap);
+            }).start())
+            {
+                Map<String, Object> jmxEnv = new HashMap<>();
+                configureClientSocketFactory(jmxEnv, encryptionOptionsMap);
+                // Invoke the same code vs duplicating any code from the JMXGetterCheckTest
+                JMXGetterCheckTest.testAllValidGetters(cluster, jmxEnv);
+            }
+        }
+    }
+
+    @Test
+    public void testClientAuth() throws Throwable
+    {
+        try(WithProperties withProperties = new WithProperties().with("javax.net.ssl.trustStore", (String)validKeystore.get("truststore"),
+                                                                      "javax.net.ssl.trustStorePassword", (String)validKeystore.get("truststore_password"),
+                                                                      "javax.net.ssl.keyStore", (String)validKeystore.get("keystore"),
+                                                                      "javax.net.ssl.keyStorePassword", (String)validKeystore.get("keystore_password"))
+        )
+        {
+            ImmutableMap<String, Object> encryptionOptionsMap = ImmutableMap.<String, Object>builder().putAll(validKeystore)
+                                                                            .put("enabled", true)
+                                                                            .put("require_client_auth", true)
+                                                                            .put("accepted_protocols", Arrays.asList("TLSv1.2", "TLSv1.3", "TLSv1.1"))
+                                                                            .build();
+
+            try (Cluster cluster = builder().withNodes(1).withConfig(c -> {
+                c.with(Feature.JMX);
+                c.set("jmx_encryption_options", encryptionOptionsMap);
+            }).start())
+            {
+                Map<String, Object> jmxEnv = new HashMap<>();
+                configureClientSocketFactory(jmxEnv, encryptionOptionsMap);
+                // Invoke the same code vs duplicating any code from the JMXGetterCheckTest
+                JMXGetterCheckTest.testAllValidGetters(cluster, jmxEnv);
+            }
+        }
+    }
+
+    @Test
+    public void testSystemSettings() throws Throwable
+    {
+        COM_SUN_MANAGEMENT_JMXREMOTE_SSL.setBoolean(true);
```

Comment:
@maulin-vasavada cant be these incorporated into WithProperties too?

Reply:
Why not! Will do

Reply:
I did it but only that method was using those variables. I still had to keep the `@After` cleanup for those variables because they are set as side-effect of the other tests in the same class.